Reference/using-ioc.md

```diff
@@ -17,128 +17,140 @@ This example will setup Autofac to work with Umbraco (see [their documentation](
-			builder.RegisterType<MyAwesomeContext>();
-
-			var container = builder.Build();
-			DependencyResolver.SetResolver(new AutofacDependencyResolver(container));
-			GlobalConfiguration.Configuration.DependencyResolver = new AutofacWebApiDependencyResolver(container);
-		}
-
-		public void OnApplicationInitialized(UmbracoApplication httpApplication, Umbraco.Core.ApplicationContext applicationContext)
-		{
-		}
-
-		public void OnApplicationStarting(UmbracoApplication httpApplication, Umbraco.Core.ApplicationContext applicationContext)
-		{
-		}
-	}
 
-In this example, we will assume that we have a Document Type called 'Home'. Now we're going to create a custom controller to hijack a route for all content pages of type Home *(NOTE: we can target custom template names too - see the [Hijacking routes](routing/custom-controllers) documentation for full details).* Notice that the constructor accepts a parameter the custom class; this will be injected via IoC.
-
-	public class HomeController : RenderMvcController
-	{
-		private readonly MyAwesomeContext _myAwesome;
+```csharp
+using Umbraco.Web;
+public class MyApplication : IApplicationEventHandler
+{
+    public void OnApplicationStarted(
+        UmbracoApplication httpApplication, Umbraco.Core.ApplicationContext applicationContext)
+    {
+        var builder = new ContainerBuilder();
+
+        // register all controllers found in this assembly
+        builder.RegisterControllers(typeof(MyApplication).Assembly);
+        builder.RegisterApiControllers(typeof(MyApplication).Assembly);
+
+        // register Umbraco MVC + web API controllers used by the admin site
+        builder.RegisterControllers(typeof(UmbracoApplication).Assembly);
+        builder.RegisterApiControllers(typeof(UmbracoApplication).Assembly);
+
+        // add custom class to the container as Transient instance
+        builder.RegisterType<MyAwesomeContext>();
+
+        var container = builder.Build();
+        DependencyResolver.SetResolver(new AutofacDependencyResolver(container));
+        GlobalConfiguration.Configuration.DependencyResolver = new AutofacWebApiDependencyResolver(container);
+    }
+
+    public void OnApplicationInitialized(UmbracoApplication httpApplication, Umbraco.Core.ApplicationContext applicationContext)
+    {
+    }
+
+    public void OnApplicationStarting(UmbracoApplication httpApplication, Umbraco.Core.ApplicationContext applicationContext)
+    {
+    }
+}
+```
 
-		public HomeController(MyAwesomeContext myAwesome)
-		{
-			_myAwesome = myAwesome;
-		}
+In this example, we will assume that we have a Document Type called 'Home'. Now we're going to create a custom controller to hijack a route for all content pages of type Home *(NOTE: we can target custom template names too - see the [Hijacking routes](routing/custom-controllers) documentation for full details).* Notice that the constructor accepts a parameter the custom class; this will be injected via IoC.
```

Comment:
You missed a note 😱 😉

Reply:
argh! I thought I was flawless... down to 99.991% 😢 

😁  - It's been fixed now 👍